### ISSUE TITLE
unable to verify the first certificate, no alarm with nocheck-tls-errors

### ISSUE THREAD

**Issue by tradexsrl** [louislam/uptime-kuma#4923]
### 📑 I have found these related issues/pull requests

i found another stale thread with no resolution

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

hi as per title if i enable no check tls error i got no alarma when the certificate is expired , if i uncheck i get unable to verify the first certificate
my cert is a paid one for our website ; i have tried to add in the enviroment as in other issue is reported:  NODE_EXTRA_CA_CERTS: /volume1/docker/uptimekuma/certs/www.crt
but when i deploy the stack i get that no extra variable are allowed.
how can i solve the problem?

### 👟 Reproduction steps

try adding the  NODE_EXTRA_CA_CERTS: /volume1/docker/uptimekuma/certs/www.crt in the stack

### 👀 Expected behavior

stack deployed

### 😓 Actual Behavior

error no extra variable allowed

### 🐻 Uptime-Kuma Version

1.23.13

### 💻 Operating System and Arch

docker

### 🌐 Browser

chrome - firefox

### 🖥️ Deployment Environment

- Runtime: docker
- Database: 
- Filesystem used to store the database on: 
- number of monitors:

**Comment by CommanderStorm** [comment]
What is the exact error message?
What is the compose file which you are submitting to docker swarm?

**Comment by tradexsrl** [comment]
i setup i website monitor and activated check of ssl cert expiration but when it check it say "unable to verify the first certificate"
i tried to check ignore ssl error but in this way i get no notification when the cert is expiring.
uptime kuma itself works perfectly , anyway this is the compose file:
version: "3.9"
services:
  uptimekuma:
    image: louislam/uptime-kuma
    container_name: Uptime-Kuma
    hostname: uptimekuma
    mem_limit: 3g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:false
    ports:
      - 3444:3001
    volumes:
      - /volume1/docker/uptimekuma:/app/data:rw

      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      TZ: Europe/Rome
    restart: always

**Comment by CommanderStorm** [comment]
`NODE_EXTRA_CA_CERTS` is not present in that file. COuld you attach the file which produces the error you noted above

> no extra variable are allowed

**Comment by tradexsrl** [comment]
hi , there is no any file that produce this error....when i add a monitor to this site www.tradexsrl.com checking to notify me if ssl is expiring , i get error each time the monitor run: unable to verify the first certificate
with this error the status page show that there is a system that is not working and we are in degraded state
i have tried to add NODE_EXTRA_CA_CERTS with no luck....from what i read i need a .pem file related to www.tradexsrl.com but i haven't a pem file , only crt .key .bundle

**Comment by CommanderStorm** [comment]
converting them to a `.pem` can be achived via oppenssl/certutil/p11-kit/...
https://stackoverflow.com/questions/991758/how-to-get-pem-file-from-key-and-crt-files

> like this:
> ```
> environment:
> NODE_EXTRA_CA_CERTS: '/volume1/docker/uptimekuma/certs/ca.crt'
> ```

Have you indented `NODE_EXTRA_CA_CERTS` properly? (where is the error coming from? is it docker or node?)

**Comment by tradexsrl** [comment]
hi , ok i have seen that my crt can be renamed to pem so i modify my compose file like this and i get no more error during deployment:
version: "3.9"
services:
  uptimekuma:
    image: louislam/uptime-kuma
    container_name: Uptime-Kuma
    hostname: uptimekuma
    mem_limit: 3g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:false
    ports:
      - 3444:3001
    volumes:
      - /volume1/docker/uptimekuma:/app/data:rw
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      UID: 1026
      GID: 100 
      TZ: Europe/Rome
      NODE_EXTRA_CA_CERTS: '/volume1/docker/uptimekuma/tradexsrl.pem'
    restart: always

i load the pem file in that directory but when i start the stack i get this error in the log:

Warning: Ignoring extra certs from `/volume1/docker/uptimekuma/tradexsrl.pem`, load failed: error:80000002:system library::No such file or directory
Welcome to Uptime Kuma
Your Node.js version: 18.20.2
2024-07-11T09:17:46+02:00 [SERVER] INFO: Welcome to Uptime Kuma

**Comment by CommanderStorm** [comment]
Please refer to the docs for volume mounting.
You are currently making the local directory `/app/data` available at `/volume1/...`. you are also allowing the container to write it.

https://docs.docker.com/storage/volumes/

> Consists of three fields, separated by colon characters (:). The fields must be in the correct order, and the meaning of each field isn't immediately obvious.
> 
> - In the case of named volumes, the first field is the name of the volume, and is unique on a given host machine. For anonymous volumes, the first field is omitted.
> - The second field is the path where the file or directory are mounted in the container.
> - The third field is optional, and is a comma-separated list of options, such as ro. These options are discussed below.

**Comment by tradexsrl** [comment]
yes i need to to do it like you explain otherwise uptimekuma cannot save the config and other files needed so when i restart the container or upgrade it , i don't lost my conf.
Do you mean that the .pem file must be placed inside the container? is not possible to place it in an external volume?
i'm asking this because to check this ssl expiration became complex to manage; first rename crt to pem , then copy in the correct position and restart the stack....in uptime kuma i have other ssl checking expiration but they are working like a charm without doing nothing

**Comment by tradexsrl** [comment]
i add it that if the pem must be placed inside the container , each time i upgrade the container image , i need to remember to redo the entire process.....

**Comment by tradexsrl** [comment]
i modified a little bit the compose and now i get no more the error of file not foud related to the pem , BUT in the monitor i get again unable to verify the first certificate

version: "3.9"
services:
  uptimekuma:
    image: louislam/uptime-kuma
    container_name: Uptime-Kuma
    hostname: uptimekuma
    mem_limit: 3g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:false
    ports:
      - 3444:3001
    volumes:
      - /volume1/docker/uptimekuma:/app/data:rw
      - /volume1/docker/certs/ca.pem:/etc/ca.pem:ro
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      UID: 1026
      GID: 100 
      TZ: Europe/Rome
      NODE_EXTRA_CA_CERTS: '/etc/ca.pem'
    restart: always

**Comment by tradexsrl** [comment]
solved: i need to rename the .bundle file to ca.pem and save it to /volume1/docker/certs/

**Comment by copenhaus** [comment]
Thanks, @tradexsrl, your last two comments solved my problem.
